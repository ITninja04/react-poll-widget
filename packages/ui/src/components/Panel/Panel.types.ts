export type PanelColor =  'blue' | 'teal' | 'pink' | 'yellow' | 'purple' | 'orange' | 'rose' | 'peach' | 'black';