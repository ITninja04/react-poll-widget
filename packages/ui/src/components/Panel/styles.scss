@import "../../colors";
@import "../../mixin";

.panel {
  @include shadow-2;
  @include roundCorners;
  position: absolute;
  display: flex;
  flex-flow: column;

  min-width: 24em;
  max-width: 24em;
  min-height: 12em;

  background: $white;
  overflow: hidden;
}
