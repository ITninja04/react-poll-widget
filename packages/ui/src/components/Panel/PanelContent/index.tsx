import React from 'react';
import cx from 'classnames';

import { PanelColor } from '../../types';

import './styles.scss';

const PanelContent: React.FC<{
  children: React.ReactNode,
  color: PanelColor
}> = ({ children, color }) => (
  <div className={cx("panel-content", color)}>
    <div className="panel-content-inner">
    {children}
    </div>
  </div>
)

export default PanelContent;