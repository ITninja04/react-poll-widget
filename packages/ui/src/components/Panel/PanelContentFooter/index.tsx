import React from 'react';
import cx from 'classnames';
import _ from 'lodash';
import { Button, Icon, Message } from 'semantic-ui-react';

import { PanelColor } from '../../types';
import './styles.scss';

const BackButton: React.FC<{
  onBack?: () => void
}> = ({
onBack
}) => (
      <Button basic icon labelPosition='left' size='tiny' onClick={onBack}>
        Back
              <Icon name='long arrow alternate left' />
      </Button>
    );

const PanelContentFooter: React.FC<{
  hasSubmit?: boolean,
  hasNext?: boolean,
  hasBack?: boolean,
  nextDisabled?: boolean,
  submitDisabled?: boolean,
  error?: string,
  onSubmit?: () => void,
  onNext?: () => void,
  onBack?: () => void,
  color?: PanelColor
}> = ({
  hasSubmit,
  hasNext = true,
  hasBack,
  nextDisabled,
  submitDisabled,
  error,
  onSubmit,
  onNext,
  onBack,
  color = 'black'
}) => (
      <div className='panel-content-footer'>
        {
          !_.isNil(error) &&
          <Message color='red'>
            <Icon name='warning circle'/>
            {error}
            </Message>
        }
        <div className={cx(
          'right-aligned row',
          { 'two-buttons': hasBack || (hasSubmit && hasNext) }
        )} >
          {
            ((hasBack && !hasNext) || (hasBack && !hasSubmit)) &&
            <BackButton onBack={onBack} />
          }
          {
            hasSubmit &&
            <Button
              disabled={submitDisabled || !_.isNil(error)}
              icon
              labelPosition='right'
              color='green'
              onClick={onSubmit}
            >
              Submit
            <Icon name='checkmark' />
            </Button>
          }
          {
            hasNext &&
            <Button
              disabled={nextDisabled || !_.isNil(error)}
              icon
              labelPosition='right'
              className={color}
              onClick={onNext}
            >
              Next
            <Icon name='long arrow alternate right' />
            </Button>
          }
        </div>
        {
          hasBack && hasSubmit && hasNext &&
          <div className='left-aligned row'>
            <BackButton onBack={onBack}/>
          </div>
        }
      </div>
    );

export default PanelContentFooter;