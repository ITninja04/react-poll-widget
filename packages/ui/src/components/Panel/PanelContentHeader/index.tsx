import React from 'react';

import './styles.scss';

const PanelContentHeader: React.FC<{ children: React.ReactNode }> = ({
  children
}) => (
    <h4 className='panel-content-header'> {children} </h4>
  );

export default PanelContentHeader;