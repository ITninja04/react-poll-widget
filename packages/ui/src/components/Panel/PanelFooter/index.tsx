import React from 'react';

import './styles.scss';

const PanelFooter: React.FC<{
  children: React.ReactNode
}> = ({ children }) => (
  <div className="panel-footer">
    {children}
  </div>
)

export const DefaultFooter = () => (
  <PanelFooter>
    Test
  </PanelFooter>
);

export default PanelFooter;