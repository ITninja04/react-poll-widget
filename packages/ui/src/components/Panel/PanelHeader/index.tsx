import React from 'react';
import cx from 'classnames';

import { PanelColor } from '../Panel.types';

import './styles.scss';

const PanelHeader: React.FC<{
  children: React.ReactNode,
  color: PanelColor
}> = ({
  children,
  color
}) => (
      <div className={
        cx(
          'panel-header',
          color
        )
      }>
        <h3>
          {children}
        </h3>
      </div>
    );

export default PanelHeader;