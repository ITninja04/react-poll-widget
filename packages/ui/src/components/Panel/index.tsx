import React from 'react';
import cx from 'classnames';
import isNil from 'lodash/isNil';

import PanelContent from './PanelContent';
import PanelHeader from './PanelHeader';
import PanelFooter, { DefaultFooter } from './PanelFooter';
import ProgressIndicator from '../ProgressIndicator';
import { PanelColor } from '../types';

import './styles.scss';

const Panel: React.FC<{
  children?: React.ReactNode;
  className?: string;
  style?: React.StyleHTMLAttributes<HTMLElement>;
  hasFooter?: boolean;
  hasHeader?: boolean;
  hasSteps?: boolean;
  footerContent?: React.ReactNode;
  headerContent?: React.ReactNode;
  color?: PanelColor;
  step?: number;
  stepsTotal?: number;
}> = ({
  children,
  className,
  style,
  hasFooter,
  hasHeader,
  hasSteps,
  footerContent,
  headerContent,
  color='blue',
  step,
  stepsTotal
}) => (
      <div
        className={cx('panel', className)}
        style={style}
      >
        {
          hasHeader &&
          <PanelHeader
            color={color}
          >
            {headerContent}
            {
              hasSteps && !isNil(step) && !isNil(stepsTotal) &&
              <ProgressIndicator
                step={step}
                stepsTotal={stepsTotal}
              />
            }
          </PanelHeader>
        }

        <PanelContent
          color={color}
        >
          {children}
        </PanelContent>

        {
          hasFooter && (
            footerContent
              ? <PanelFooter>
                {footerContent}
              </PanelFooter>
              : <DefaultFooter />
          )
        }
      </div>
    )

export default Panel;