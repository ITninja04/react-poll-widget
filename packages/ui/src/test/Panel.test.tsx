import Panel from "../components/Panel";
import { createSnapshotTest } from "./utils";

createSnapshotTest(Panel, {}, 'Empty panel')
createSnapshotTest(Panel, { hasFooter: true }, 'Panel with footer')
createSnapshotTest(
  Panel, 
  {
    color: 'blue',
    hasHeader: true, 
    hasFooter: true,
    headerContent: 'Snapshot test header'
  }, 
  'Panel with header'
  )
createSnapshotTest(
  Panel,
  {
    color: 'blue',
    hasHeader: true,
    hasFooter: true,
    hasSteps: true,
    headerContent: 'Snapshot test header',
    step: 2,
    stepsTotal: 5
  },
  'Panel with step indicator'
)
