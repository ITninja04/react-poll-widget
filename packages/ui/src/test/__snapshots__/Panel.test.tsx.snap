// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Snapshot) Empty panel should render 1`] = `
<DocumentFragment>
  <div
    class="panel"
  >
    <div
      class="panel-content blue"
    >
      <div
        class="panel-content-inner"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`(Snapshot) Panel with footer should render 1`] = `
<DocumentFragment>
  <div
    class="panel"
  >
    <div
      class="panel-content blue"
    >
      <div
        class="panel-content-inner"
      />
    </div>
    <div
      class="panel-footer default-footer"
    >
      This poll widget was created by
      <a
        href="https://github.com/nukeop/"
      >
        nukeop
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`(Snapshot) Panel with header should render 1`] = `
<DocumentFragment>
  <div
    class="panel"
  >
    <div
      class="panel-header blue"
    >
      <h3>
        Snapshot test header
      </h3>
    </div>
    <div
      class="panel-content blue"
    >
      <div
        class="panel-content-inner"
      />
    </div>
    <div
      class="panel-footer default-footer"
    >
      This poll widget was created by
      <a
        href="https://github.com/nukeop/"
      >
        nukeop
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`(Snapshot) Panel with step indicator should render 1`] = `
<DocumentFragment>
  <div
    class="panel"
  >
    <div
      class="panel-header blue"
    >
      <h3>
        Snapshot test header
        <div
          class="progress-indicator"
        >
          <svg>
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <circle
              cx="12"
              cy="12"
              r="10"
              stroke-dasharray="62.830000000000005"
              stroke-dashoffset="37.698"
            />
          </svg>
        </div>
      </h3>
    </div>
    <div
      class="panel-content blue"
    >
      <div
        class="panel-content-inner"
      />
    </div>
    <div
      class="panel-footer default-footer"
    >
      This poll widget was created by
      <a
        href="https://github.com/nukeop/"
      >
        nukeop
      </a>
    </div>
  </div>
</DocumentFragment>
`;
